### ISSUE TITLE
Fix S3253 FP:  Don't raise for primary constructor in type declarations without parameters when they inherit from types with parameters

### ISSUE THREAD

**Issue by thomasfroehle** [SonarSource/sonar-dotnet#8436]
### Description

S3253 is reported when a record with a parameterless constructor inherits another record that has parameters and passes them inline (instead of having an explicit base()-call).

### Repro steps

```
public abstract record Base(string Value);

public record Foo() : Base("SomeValue");
```

### Expected behavior

No S3253 is raised because the Foo()-constructor is not redundant. 

### Actual behavior

S3253 is raised.

### Known workarounds

```
public abstract record Base(string Value);

public record Foo : Base
{
    public Foo() : base("SomeValue")
    {

    }
}
```

### Related information

SonarAnalyzer.CSharp: 9.14.0.81108

**Comment by gregory-paidis-sonarsource** [comment]
Hey there!
I confirm that this is indeed a false positive.
I made a [reproducer](https://github.com/SonarSource/sonar-dotnet/pull/8454) to track this, feel free to take a look.

Just so you know, this rule is not part of the default, SonarWay profile, so updates to it will come a bit slower than SonarWay rules.
Until this is fixed, feel free to disable the rule from your profile if it is becoming too noisy for you. :)

**Comment by thomasfroehle** [comment]
Thanks for the feedback. The repro fits perfect to the issue.

**Comment by DominikAmon** [comment]
Please note, that is also affects regular classes as well (not only records)

Example:
```
public class ClaimsIdentityExpectedException() : CodeException(Properties.Resources.TtWkisEx0001ClaimsIdentityExpectedException,Properties.Resources.TtWkisEx0001ClaimsIdentityExpectedCode)
{
}
```
